### PR TITLE
[CSL-2238] Bundle

### PR DESCRIPTION
**Pull request by esezen**
Ways to use:

```js
window.ConstructorioQuizzes({
  selector: "#app",
  includeCSS: false,
  apiKey: "",
  quizId: "coffee-quiz",
  resultsPageOptions: {
    addToCartCallback: (item) => console.dir(item),
    resultCardRegularPriceKey: "price",
    resultCardSalePriceKey: "salePrice"
  }
});
```

```js
import CioQuiz from "./constructorio-ui-quizzes-bundled";

CioQuiz({
  selector: "#app",
  includeCSS: false,
  apiKey: "",
  quizId: "coffee-quiz",
  resultsPageOptions: {
    addToCartCallback: (item) => console.dir(item),
    resultCardRegularPriceKey: "price",
    resultCardSalePriceKey: "salePrice"
  }
});
```

### CODE REVIEW

**Review by jjl014 — COMMENTED** [review]
Nice work! This is looking great from what I can tell! 🚀 

Left a couple of comments.

File: src/bundled.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          <CioQuiz
+            quizId={quizId}
+            apiKey={apiKey}
+            cioJsClient={cioJsClient}
+            resultsPageOptions={resultsPageOptions}
+            quizVersionId={quizVersionId}
+          />
+        </React.StrictMode>
+      );
+    }
```
[the remark this answers is not in the document]
Comment (esezen) [reply src/bundled.jsx:41]: Good idea. I added an error message. Feel free to edit the message directly if you can think of a better one

File: vite.config.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    styleId: 'cio-quizzes-styles'
+  })],
+  build: {
+    rollupOptions: {
+      preserveEntrySignatures: 'strict',
+      input: {
+        app: "./src/bundled.jsx",
+      },
+      output: {
+        sourcemap: false,
```
[the remark this answers is not in the document]
Comment (esezen) [reply vite.config.js:15]: Good catch

File: src/bundled.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  apiKey,
+  quizId,
+  cioJsClient,
+  resultsPageOptions,
+  quizVersionId,
+  includeCSS = true,
+}) => {
+  if (document) {
+    if (!includeCSS) {
+      try {
```
[the remark this answers is not in the document]
Comment (esezen) [reply src/bundled.jsx:17]: I added it last minute just being paranoid. I wanted to make sure we don't error out in any circumstance but you are right it's unnecessary

File: src/bundled.jsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  includeCSS = true,
+}) => {
+  if (document) {
+    if (!includeCSS) {
+      const stylesheet = document.getElementById('cio-quizzes-styles');
+
+      if (stylesheet) {
+        stylesheet.disabled = true;
+      }
+    }
```
Comment (jjl014) [inline src/bundled.jsx:22]: Forgot to mention, but if we're doing it this way, should we also have an else statement here with:

`document.getElementById('cio-quizzes-styles').disabled = false`

**Review by jjl014 — APPROVED** [review]
Thanks for making the changes. This looks great to me! 🚀

**Comment by esezen** [comment]
Enes to update README with bundled approach examples

File: README.md [hunk 5]
```diff
@@ -33,6 +35,26 @@ function YourComponent() {
     </div>
   );
 ```
+### Import Javascript Bundle
+
+This is a framework agnostic method that can be used in any JavaScript project. The `CioQuiz` function will mount a Quiz to the element for the given selector and it will handle state management, data fetching, and rendering logic for the entire quiz.
```
Comment (esezen) [inline README.md:40]: I don't love

> The `CioQuiz` function will mount a Quiz to the element for the given selector and it will handle state management, data fetching, and rendering logic for the entire quiz.

but I couldn't find anything better. @jjl014 feel free to edit directly

File: src/bundled.jsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    ReactDOM.createRoot(containerElement).render(
+      <React.StrictMode>
+        <CioQuizComponent {...rest} />
+      </React.StrictMode>
+    );
+  }
+};
+
+if (window) {
+  window.CioQuiz = CioQuiz;
```
Comment (esezen) [inline src/bundled.jsx:35]: Decided to make this `CioQuiz` to make it consistent with the component name

File: src/bundled.jsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+      if (!includeCSS) {
+        stylesheet.disabled = true;
+      } else {
+        stylesheet.disabled = false;
+      }
+    }
+
+    ReactDOM.createRoot(containerElement).render(
+      <React.StrictMode>
+        <CioQuizComponent {...rest} />
```
Comment (esezen) [inline src/bundled.jsx:28]: Added `{...rest}` here so we don't forget to add new props to this when we make changes to the component

File: README.md (hunk 5 above)
Comment (jjl014) [reply README.md:40]: Maybe we could re-use the description from our docs:

> Allows for a simple interface to inject a quizzes UI into the DOM directly simply by supplying a container reference.

But reword it to fit a bit better for the repo:

> The `CioQuiz` function provides a simple interface to inject an entire Quizzes UI into the provided `selector`.

Probably no necessary to mention state management, data fetching, etc here. Maybe it could be something we can add to the library description at the top of the README instead

File: README.md [hunk 8]
```diff
@@ -33,6 +35,26 @@ function YourComponent() {
...
+In addition to [Quiz component props](https://constructor-io.github.io/constructorio-ui-quizzes/?path=/docs/quiz-component--docs), this function also accepts `selector` and `includeCSS`.
+
+```js
+import CioQuiz from "@constructor-io/constructorio-ui-quizzes/constructorio-ui-quizzes-bundled";
+
+CioQuiz({
+  selector: "#quiz-container",
+  quizId: "coffee-quiz",
+  apiKey: "key_wJSdZSiesX5hiVLt",
+  includeCSS: true,
```
Comment (jjl014) [inline README.md:50]: Should we add in a little comment here to explain what this does? For example:

```
includeCSS: true, // Include the default CSS styles. Defaults to true.
```

File: src/bundled.jsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+import CioQuizComponent from './components/CioQuiz';
+import './styles.css';
+
+const CioQuiz = ({ selector, includeCSS = true, ...rest }) => {
+  if (document) {
+    const stylesheet = document.getElementById('cio-quizzes-styles');
+    const containerElement = document.querySelector(selector);
+
+    if (!containerElement) {
+      console.error(`ConstructorioQuizzes: There were no elements found for the provided selector`);
```
Comment (jjl014) [inline src/bundled.jsx:13]: Let's update this to `CioQuiz: There were....` since we're changing the window var

File: src/bundled.jsx (hunk 7 above)
Comment (jjl014) [reply src/bundled.jsx:28]: Would there be any issues if the consumer passes in random "props"? 🤔

**Review by jjl014 — COMMENTED** [review]
Nice, thanks for the updates!

Left some comments and questions.
